src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -953,6 +953,7 @@ BlockIO InterpreterCreateQuery::createTable(ASTCreateQuery & create)
 
         // Table SQL definition is available even if the table is detached (even permanently)
         auto query = database->getCreateTableQuery(create.getTable(), getContext());
+        FunctionNameNormalizer().visit(query.get());
```

Comment:
Looks safe, but let's check what we store in ZooKeeper for Replicated tables. Do we normalize values there?

Reply:
This PR（https://github.com/ClickHouse/ClickHouse/pull/33933） is already compatible with the non-standardized metadata stored by ZK